packages/health-check/src/checks/critical.js

```diff
@@ -60,7 +86,44 @@ async function uploadCheck(done) {
     data.ip = error?.response?.ip ?? null;
   }
 
-  done({ name: "upload_file", time: calculateElapsedTime(time), ...data });
+  done({ name, time: calculateElapsedTime(time), ...data });
+}
+
+// skylinkHealthCheck checks if the skylink has reached full redundancy
+async function skylinkHealthCheck(skylink, numRetries = 30, authCookie, isLarge = false) {
+  // Get the health of the skylink
+  const response = await got(`https://${process.env.PORTAL_DOMAIN}/skynet/health/skylink/${skylink}`, {
+    headers: { cookie: authCookie },
+  });
+  const healthData = getResponseContent(response);
+
+  // Check Basesectorredundancy first
+  if (healthData.basesectorredundancy !== 10 && numRetries > 0) {
+    // Semi-smart sleep before retrying. Sleep longer if the redundancy is
```

Comment:
so 10 factorial is (n^2 + n)/2 or whatever so that's ~55s, is that enough? I guess it should be but it's not a huge amount of time either right

Reply:
It should be. If we are failing this I think is it just cause for us to investigate. Our servers have 1gbps connections and streaming upload should be priority 1 for workers so there shouldn't be anything blocking this from uploading 40MB in 55s.